### ISSUE TITLE
Feature Requests

### ISSUE THREAD

**Issue by Vdauphin** [Vdauphin/HeartsAndMinds#738]
### **Before reporting**

-  Any and all requests should be done here.
- Before posting a new one, please make sure to check the previous entries on the thread and make a quick search; please don't reiterate requests that had already been either accepted for inclusion or disregarded.
- Accepted feature requests may be moved to a separate issue by moderators for further discussion.

**Comment by GoldJohnKing** [comment]
Suggestion for suicide car bombs:
Spawn a suicider, then order him to get in a vehicle. This method is functionable to suiciders spawned by Achilles mod and ACE3 mod, so I wonder if it's implementable in a mission.

**Comment by Vdauphin** [comment]
Any work on suicide car bomb should take care of this: https://github.com/Vdauphin/HeartsAndMinds/issues/347#issuecomment-328270613

**Comment by GoldJohnKing** [comment]
[Question]
Is there any condition for FOB deploy action to be shown?
I'm using unchanged master_daily branch and there's no FOB deploy action no matter where I put the blue container.

**Comment by Vdauphin** [comment]
Nop that a bug!

**Comment by GoldJohnKing** [comment]
Hi, the big shower is too easy to flip if a vehicle knock on it. Would you make it solid?

**Comment by Vdauphin** [comment]
Already tried but nothing :/
I was thinking to use solid barrier around but I prefer let admin customize it

**Comment by GoldJohnKing** [comment]
Set the shower to "enableSimulation false" might help, but the animation would be disabled then.

enableSimulation  usage on wiki pages:
https://community.bistudio.com/wiki/enableSimulation
https://community.bistudio.com/wiki/enableSimulationGlobal

**Comment by GoldJohnKing** [comment]
Another question: Has all objects genetation (including vehicles, buildings, items and may be units) 
already moved to "align to ground surface"? I've recently noticed some chemical warfare items (like the container wreck and the big tent) and enemy's heavy ground vehicles (like tanks or BTRs) are generated half-undergrounded (sometimes vehicles may bounce) in very steep areas.

**Comment by Vdauphin** [comment]
> Set the shower to "enableSimulation false" might help, but the animation would be disabled then.

This was the first thing I tried

>  enemy's heavy ground vehicles (like tanks or BTRs)

Yes (https://github.com/Vdauphin/HeartsAndMinds/blob/master/=BTC=co@30_Hearts_and_Minds.Altis/core/fnc/mil/createVehicle.sqf#L48)

> (like the container wreck and the big tent)

Yes (https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/common/create_composition.sqf#L44)

From database https://github.com/Vdauphin/HeartsAndMinds/pull/448#issuecomment-352109721

The issue could still happen if 1 meter away the terrain change of shape. Create an issue if you have more intel 

Cheers

**Comment by CurePindal** [comment]
Suggestion: Support for AI toons
Hello! Havent found any way for AI soldiers in the missions. We would love to have at least some basic support for AIs, because we usually play as a very small group, which isnt exactly easy to handle in H&Ms. Even just basic support for AI would be great! Would also enable being able to play solo.
 thx for considering. Keep up the great work!

**Comment by Vdauphin** [comment]
Hello, for now you could use Zeus
Have fun!

**Comment by Sir-Stone** [comment]
Any plans on re-adding the interpreter role or has another role taken that position?

**Comment by Vdauphin** [comment]
The interpreter role never gone ... (?)

**Comment by Sir-Stone** [comment]
> The interpreter role never gone ... (?)
> Look here: https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/mission.sqm#L1034

Appreciate it.

**Comment by mrNo0b** [comment]
To fix the big shower flipping problem add this code to the black tarp under the shower init:
**`btc_bigShower attachTo [this];`**

**Comment by Vdauphin** [comment]
Thanks for this suggestion, sadly the animation is affected :/
But it is definitely a good work around

**Comment by mrNo0b** [comment]
I did it and the animation was not affected, make sure you have `enable simulation` **on** for the shower, also disable damage for both

**Comment by Vdauphin** [comment]
Continuous spray without `attachTo`:
![20200414133457_1](https://user-images.githubusercontent.com/14364400/79220749-e7d3ee00-7e54-11ea-903c-252a5bf5ba55.jpg)
Bubble spray with `attachTo`:
![20200414133634_1](https://user-images.githubusercontent.com/14364400/79220808-0508bc80-7e55-11ea-8ed4-d17d164e5815.jpg)

But it is a good work around.

**Comment by mrNo0b** [comment]
- Is it possible to add option to redeploy from FOB to base and other fobs?
I added this feature but for some reason its impossible to redeploy to some fobs, screen goes black and then player remain at the same place instead of teleporting
- Is it possible to spawn empty civilian vehicles at towns when players enter?
- Is it possible to add markers to base vehicles so players will be able to see them on map?
I used scripts to add markers and civ veh spawn, but maybe there are better ways to do it
- also I changed the fob structure to `Land_Cargo_House_V4_F` because its not realistic to spawn huge 2 store building from single shipping container, btc_fob_mat I set to `B_Slingload_01_Medevac_F`

**Comment by Vdauphin** [comment]
> Is it possible to add option to redeploy from FOB to base and other fobs?
I added this feature but for some reason its impossible to redeploy to some fobs, screen goes black and then player remain at the same place instead of teleporting

With a first look I don't see why. I am working on something more standard: https://github.com/Vdauphin/HeartsAndMinds/pull/825

>Is it possible to spawn empty civilian vehicles at towns when players enter?

You could add civilian vehicles class name here: https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/def/mission.sqf#L299
But remember those objects spawn as super simple objects for performance reason

> Is it possible to add markers to base vehicles so players will be able to see them on map?
No, generally I dislike this but optional is okay, pull request is welcome...

**Comment by mrNo0b** [comment]
Didn't wanted to open new issue as I am not that good sqf programmer so maybe I didn't understood this line: https://github.com/Vdauphin/HeartsAndMinds/blob/a332b7c7b9480e921a521ad2b4984b384f6a082e/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/fob/redeploy.sqf#L60 is it possible that it can return `<null>` as _pos?
I have issue with some fobs redeploy not working, and I think I managed to fix it by changing this line to: `selectRandom _positions;`

**Comment by Vdauphin** [comment]
This line search for positions not on roof (`_x select 2 < 1`).

> is it possible that it can return <null> as _pos?

I don't think so but may be if your FOB have only position on roof, I will check tomorrow with your `Land_Cargo_House_V4_F`

**Comment by GoldJohnKing** [comment]
Your discussion makes me come up with a question:
How to decontaminate a heli? Use a truck to tow it?

**Comment by Vdauphin** [comment]
> How to decontaminate a heli?

You need to destroy it and repair the wreck if it is not a helo. This could be done if the big shower is bigger...

> Use a truck to tow it?

You can't tow heli

**Comment by mrNo0b** [comment]
Suggestions:
- Add `ACE_SelfActions` to Officer that will allow selecting specific side mission type and random.
- Add "traitor" civilian that will give Intel that will be very accurate but when players will go to the marker it will spawn waves of enemies that will block all exits, ambush area should be in towns or areas with cover, not open fields, maybe show notification/set objective to players saying "Survive!"
- or add chance that side mission will be ambush instead, so when players go to kill general for example they will get in ambush and side mission objective will change to "Survive".

**Comment by NZF-JDWang** [comment]
This could be an engine limitation, but it would be cool if we could use the flatbed HEMTT to bring wrecks back to base. 
I know you can load a hunter on the back, but pretty sure you can't load a wreck. (thanks BI) 
Maybe instead of using the ViV function you could do an attachTo?

Although I suppose with all the different models from all the different mods people use it would be difficult to script

**Comment by GoldJohnKing** [comment]
> it would be cool if we could use the flatbed HEMTT to bring wrecks back to base.

This is quite easy.

You can modify https://github.com/Vdauphin/HeartsAndMinds/blob/4041d163ef78f787d82def06db2fc7745bd5b905/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/eh/veh_init.sqf to implement the feature.

Cheers.

**Comment by mrNo0b** [comment]
I think NZF-JDWang want to use the flatbed HEMTT to attach the wrecks/objects on the flatbed itself instead of towing it, I think we need this feature

**Comment by NZF-JDWang** [comment]
@GoldJohnKing Thanks but @mrNo0b was correct, I know how to set up the towing etc, but I was hoping there could be some way to load wrecks onto the flatbed. 

I don't want to tow it, I want to put it on the bed of the truck and drive back to base like that.

**Comment by GoldJohnKing** [comment]
> I don't want to tow it, I want to put it on the bed of the truck and drive back to base like that.

Ah I get it. I vote for the feature, too, cause it looks better than towing. 😅

**Comment by GoldJohnKing** [comment]
H&M used to keep player's loadouts when player die then respawn. However, after use BI Respawn, players always respawn with default loadouts and does not follow "remove weapons" params either. I'd like to have this feature back...

**Comment by Vdauphin** [comment]
> H&M used to keep player's loadouts when player die then respawn.

Yes, this is possible because in mission.sqm the CBA setting `ace_respawn_savepredeathgear` is ON.

> However, after use BI Respawn, players always respawn with default loadouts

You should respawn with the loadout you have before respawn thanks to CBA setting `ace_respawn_savepredeathgear`.

> and does not follow "remove weapons" params either. 

If you are talking about the https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/def/param.hpp#L314-L318 parameter, when it is set on "Remove all weapons", the player spawn without weapons when connecting but does not apply on respawn.

I am not sure to understand well where the behavior changed since BI Respawn :/

**Comment by GoldJohnKing** [comment]
Nope. None of those behavior works any more...
I set ace_respawn_savepredeathgear on, and set the "Remove all weapons" param, players still respawn with default loadouts (defined in eden editor).

**Comment by canprod** [comment]
When we talk to a civilian, he can ask for help.
Like:
-heal his friends in the next town
-give food to the population
-kill terrorist in the house
-free his friend

All the mission need to be near the civilian.
All the mission make reputation and the civilian can give more precise hint for the hideout or cache
thats make the mission even more immersive.

**Comment by Vdauphin** [comment]
> the civilian can give more precise hint for the hideout or cache thats make the mission even more immersive.

This is already the case: https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/info/ask.sqf#L90-L99

**Comment by canprod** [comment]
yes but after we complete his mission we can ask again and its more precise

**Comment by GoldJohnKing** [comment]
I got an idea about that.

We can have a parameter titled "Need to contact civilian to gather mission detail" with "Always", "Random" and "Never" selections.

When set to "Always", the new generated mission title would be "Civilian ask for help, please go and check", then players need to find and contact the civilian to reveal mission detail. And after finished the job, we can return to the same people for an upvote on our overall reputation or request some intel (fake or real based on overall reputation but finish his job would increase the real probability). Let player choose what to do.

Those revealed mission types could not only be limited to existing mission types, but also include some intel, such as cache position and enemy activity location and so on...

Besides, fake request could also be generated based on reputation, such as enemy hidden deeply inside the people's houses, or when player contact him, civilians in the city take out guns and fight against players... Still, a parameter titled "Fake civilian request enabled" with "Always", "Random" and "Never" selections could be added.

**Comment by mrNo0b** [comment]
essence of Arma is bugs, glitches and low fps

**Comment by GoldJohnKing** [comment]
Hi, is there any way to spawn a FOB at mission start, or set a structure as FOB or permanent redeploy position in eden editor, so that player can teleport among the two or more position without manually deploying a FOB?

**Comment by Vdauphin** [comment]
Hello,
> is there any way to spawn a FOB at mission start,

Add this:
```sqf
[[1,1,1], 0, "My FOB"] call btc_fnc_fob_create_s;
```
here: https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/init_server.sqf#L20

See this for parameters: https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/fob/create_s.sqf#L2-L27

Cheers

**Comment by GoldJohnKing** [comment]
Feature Request: Add support for UAVs respawn (including recovering from wreck).
Currently, "Virtual crew" in UAV does not respawn no matter if it's set to be `btc_helo` or `btc_veh`, so UAV like `B_UAV_05_F `and `B_UAV_02_F` can be used only once.
I remember using ACE interaction could add crew into UAV, but it seems not working now. So could it be a bug?

**Comment by Vdauphin** [comment]
> Currently, "Virtual crew" in UAV does not respawn no matter if it's set to be btc_helo or btc_veh,

`btc_vehicles` already have this feature, https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/log/createVehicle.sqf#L53-L55

`btc_helo` don't have it https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/eh/veh_respawn.sqf

> So could it be a bug or did I miss something?

it is a bug :p

**Comment by GoldJohnKing** [comment]
Hi, could you please add a feature to delete dead bodies, vehicle wrecks (exclude btc_veh) and ground weapon holders near the base and making those garbage only exists for a certain time?

Because of the BI respawn/redeploy function, many more players choose to respawn instead of interact with the Arsenal to redeploy to another location, cause great amount of bodies and really low frame rate at base.

Because of BI cleanup is disabled, those bodies would exists forever unless Zeus delete them manually.

I think this could be a critical problem.

**Comment by Vdauphin** [comment]
> Hi, could you please add a feature to delete dead bodies, vehicle wrecks (exclude btc_veh) and ground weapon holders near the base and making those garbage only exists for a certain time?

This is already available: https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/city/cleanUp.sqf

> Because of the BI respawn/redeploy function, many more players choose to respawn instead of interact with the Arsenal to redeploy to another location, cause great amount of bodies and really low frame rate at base.

This "respawn" should be named "suicide" as it kill player and decrease reputation

**Comment by GoldJohnKing** [comment]
>This is already available: https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/city/cleanUp.sqf

I've read that code, btc_fnc_city_cleanUp seems only called when a city is deactivated, it does not always clean up garbage at base.

> This "respawn" should be named "suicide" as it kill player and decrease reputation

Yes, I've noticed that... new players just wont listen to anything...

**Comment by Vdauphin** [comment]
> Yes, I've noticed that... new players just wont listen to anything...

Do your admin job: https://community.bistudio.com/wiki/Description.ext#respawnButton

**Comment by GoldJohnKing** [comment]
> > Yes, I've noticed that... new players just wont listen to anything...
> 
> Do your admin job: https://community.bistudio.com/wiki/Description.ext#respawnButton

This can be used to prevent player from manually respawn, but would also cause unconscious player unable to respawn either, and player cannot recover from bug by respawn, which could be a significant side effect to players.

Besides, this does not solve the rubbish around base issue. Sometimes enemy continuously goes to and attack player base, so I think a garbage collect logic for player base would be a better solution.

**Comment by mrNo0b** [comment]
I added trigger to base and made it activate when no players are around, it will remove all destroyed objects, dropped gear, dead bodies (I think)
add this to Activated code of the trigger (in my case I have this code in function):

	// delete all destroyed player placed objects
	if (count btc_log_obj_created > 0) then {
		{
			if (damage _x == 1) then { _x call CBA_fnc_deleteEntity; };
		} forEach btc_log_obj_created;

		btc_log_obj_created = btc_log_obj_created - [objNull];
	};

	// clear all gear from ground
	private _objs = nearestObjects [getPos nc_trig_baseCleaner, ["GroundWeaponHolder"], 500]; 
 
	if (count _objs > 0) then {
		{
			_x call CBA_fnc_deleteEntity;
		} foreach _objs;
	};

	if (btc_debug) then {
		["[!] Base cleaner activated"] remoteExec ["systemChat", -2];
	};

trigger name is **nc_trig_baseCleaner** but if you use this code inside trigger replace it with **this**, maybe this code will miss some items, but it worked fine when I tested it on local dedicated server...
I agree with GoldJohnKing, there should be "base cleaner" by default.
server that I used to play this mission on switched to another one so now I can't edit it.

**Comment by GoldJohnKing** [comment]
Thanks to @mrNo0b , I come up with an idea.

Just use this inside the trigger:
```
[btc_base] call btc_fnc_city_cleanUp;
if (btc_debug) then {
    ["Base Cleanup Triggered."] remoteExec ["systemChat", 0];
};
```

When trigger activates, the `btc_fnc_city_cleanUp` is called, and garbage within certain range of `btc_base` would be deleted, and this is fully compatible with the mission, without any need to think about item types and `btc_veh` or something.

**Comment by canprod** [comment]
New interaction with civilian " Give food" Search in inventory to give water or food
Physical intel spawn random in house

New effect with hight or low reputation:

low:
- door can be close
- all civilian in city take weapon

hight : 
- random event in city ( barbecu, party)

**Comment by RoqueTHEGAMER** [comment]
Ability to remove the rubbish with the shovel that hide the IED in the middle of the road after disarming an IED to help clear the road and also give it a small amount of reputation without needing the task to do it.

And for the remove the rubbish task, instead of ask for the Nemmera, ask the use of the shovel from ace (Entrenching tool) since it's a little immersion breaking having to use that vanila vehicle depending on the faction, and also, such small rubbish can be dealt even with bare hands

**Comment by RoqueTHEGAMER** [comment]
Additionally, make the rubbish be able to be removed without the shovel and make the action slower if is done without it.
Here is my solution that I placed in onPlayerRespawn.sqf since you can't add action to simple objects
```
[
	player,											// Object the action is attached to
	"Remove Rubbish",										// Title of the action
	"\a3\missions_f_oldman\data\img\holdactions\holdAction_box_ca.paa",	// Idle icon shown on screen
	"\a3\missions_f_oldman\data\img\holdactions\holdAction_box_ca.paa",	// Progress icon shown on screen
	"((getModelInfo cursorObject select 1) in btc_model_ieds) and {_this distance cursorObject < 3}",						// Condition for the action to be shown
	"_caller distance cursorObject < 3",						// Condition for the action to progress
	{
		if ("ACE_EntrenchingTool" in (Items player)) then {_duration = 5};
	},													// Code executed when action starts
	{},													// Code executed on every progress tick
	{deleteVehicle cursorObject},				// Code executed on completion
	{},													// Code executed on interrupted
	[],													// Arguments passed to the scripts as _this select 3
	12,													// Action duration [s]
	10,													// Priority
	false,												// Remove on completion
	false												// Show in unconscious state
] call BIS_fnc_holdActionAdd;
```

**Comment by RoqueTHEGAMER** [comment]
Feature request: make btc_base already the default safe area in [] call btc_fnc_city_init;

in line 64 to 67
```sqf
 if ((getMarkerPos "btc_base") inArea [_position, 500, 500, 0, false]) exitWith {};
```

**Comment by JWB502** [comment]
> Feature request: make btc_base already the default safe area in [] call btc_fnc_city_init;
> 
> in line 64 to 67
> 
> ```sqf
>  if ((getMarkerPos "btc_base") inArea [_position, 2000, 2000, 0, false]) exitWith {};
> ```

I'd have to agree with this one as well. Maybe not 2000m but the concept, absolutely!

**Comment by canprod** [comment]
different type of hideout:
1 small: difficult to find, without the raid flag, small box, sleeping bag
1 medium: big box, no red flag, campfire, some lights.
1 big: big box redflag easy to find with binoculare,

settings in param.hpp like IED

**Comment by SgtKrunch** [comment]
> Already tried but nothing :/
> I was thinking to use solid barrier around but I prefer let admin customize it

Random idea, would using like attachTo to something like an H-Barrier prevent it from being knocked over while maintaining the animation?

**Comment by GoldJohnKing** [comment]
Hi, I wonder if you could add a "pre-spawned city" function and corresponding parameters in param.hpp.

"pre-spawn" means regardless of players' approch, the city will always be active while certain condition(s) are met:
1. City with hideout should be pre-spawned.
2. City with cache should be pre-spawned.
3. City with active side-missions.
4. All above.

This aims to solve the issue that when player performs a "fast land" into the city center from helicopter, enemy AI would spawn right in the face of the players, as well as to make air observation available even no player is around the city.

Also, there can also be a function to configure which city can be active, for example, only city with hideout, cache or side-missions or cities around them can be active.

This is because in most conditions, players would aim at a certain AO of the whole terrain. This can prevent cities passed by players from activating, which may improve performance when there are plenty of players all around the map driving on their way to AO, and make it easier for players to get to the AO by land vehicles.

**Comment by Raz0rLegend** [comment]
I thought about some poster spawning on the wall inside the cities with an interaction ace to tear off. This action could give us some reputation. This way the ENI renforcement will be weaker.

**Comment by Tetlys** [comment]
Would be great to see alternative ways to breach doors, A breaching charge type situation (At risk of damaging the building ofc) Maybe an event handler to look for explosions near doors?

**Comment by Tetlys** [comment]
Also, The same code applied to Vehicle towing works with ace cargo containers, I was able to change the container types to Huron Containers and load them neatly onto flatbeds. Our players use it as the primary method of transport :)

**Comment by Raz0rLegend** [comment]
I had an idea too. I would be to great to create a Maximum capacity off tickets.
We tried to launch the new version on PO and the guys tried to rush the tickets gameplay so now, we've got 60 tickets, and right now it become useless.

Second thing we should talk to it in vocal but, it seems to me it will be a nice thing to Add the Liberation Capacity to reach some furnitures like Fuel/Ammo/etc To get news fobs or news veh. Like this way, we could start with less veh. And the base with the furnitures and the reputation could allow us certain veh, and the numbers of them.

What do you think about it ?

**Comment by UncreatedLemon1** [comment]
Loading and saving of vehicle pylons able to be implemented to the save system?

https://community.bistudio.com/wiki/Arma_3:_Vehicle_Loadouts

Looks like it could be handy, happy to see if I can figure a way to bring it to the mission file

**Comment by GoldJohnKing** [comment]
> Loading and saving of vehicle pylons able to be implemented to the save system? --> https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/db/load.sqf#L156
> 
> https://community.bistudio.com/wiki/Arma_3:_Vehicle_Loadouts
> 
> Looks like it could be handy, happy to see if I can figure a way to bring it to the mission file

This has be done with ACE Pylon framework.

**Comment by UncreatedLemon1** [comment]
> > Loading and saving of vehicle pylons able to be implemented to the save system? --> https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/db/load.sqf#L156
> > 
> > https://community.bistudio.com/wiki/Arma_3:_Vehicle_Loadouts
> > 
> > Looks like it could be handy, happy to see if I can figure a way to bring it to the mission file
> 
> This has be done with ACE Pylon framework.

When I save and restart the server, the Pylons reset back to default state, this is with 3cb factions mod...do you see this occur too?

**Comment by GoldJohnKing** [comment]
> > > Loading and saving of vehicle pylons able to be implemented to the save system? --> https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/db/load.sqf#L156
> > > https://community.bistudio.com/wiki/Arma_3:_Vehicle_Loadouts
> > > Looks like it could be handy, happy to see if I can figure a way to bring it to the mission file
> > 
> > 
> > This has be done with ACE Pylon framework.
> 
> When I save and restart the server, the Pylons reset back to default state, this is with 3cb factions mod...do you see this occur too?

Use `vehicles` instead of `respawnable_vehicles` would save the pylon.

**Comment by UncreatedLemon1** [comment]
> > > > Loading and saving of vehicle pylons able to be implemented to the save system? --> https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/db/load.sqf#L156
> > > > https://community.bistudio.com/wiki/Arma_3:_Vehicle_Loadouts
> > > > Looks like it could be handy, happy to see if I can figure a way to bring it to the mission file
> > > 
> > > 
> > > This has be done with ACE Pylon framework.
> > 
> > When I save and restart the server, the Pylons reset back to default state, this is with 3cb factions mod...do you see this occur too?
> 
> Use `vehicles` instead of `respawnable_vehicles` would save the pylon.

Vehicles are not set to respawn, however, we do spawn them in using admin login point

**Comment by Vdauphin** [comment]
> > > > > Loading and saving of vehicle pylons able to be implemented to the save system? --> https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/db/load.sqf#L156
> > > > > https://community.bistudio.com/wiki/Arma_3:_Vehicle_Loadouts
> > > > > Looks like it could be handy, happy to see if I can figure a way to bring it to the mission file
> > > > 
> > > > 
> > > > This has be done with ACE Pylon framework.
> > > 
> > > 
> > > When I save and restart the server, the Pylons reset back to default state, this is with 3cb factions mod...do you see this occur too?
> > 
> > 
> > Use `vehicles` instead of `respawnable_vehicles` would save the pylon.
> 
> Vehicles are not set to respawn, however, we do spawn them in using admin login point

H&M save pylon as stated and you just have to use https://ace3mod.com/wiki/feature/pylons.html#1-overview to change them in-game

**Comment by UnRealxInferno** [comment]
Add BRF Factions support,

Perhaps add saving of vehicle contents/condition and crate contents, player hunger/thirst (ace), and medical condition (if it isn't saved already)

**Comment by arma-fury-road** [comment]
### Side Mission Idea: related to a Workshop mod


* **[+]** Brings a freshness and a renewal with the contribution of new elements in the side missions and sets up a collaboration with a creator of the Workshop.
* **[-]** Brings a dependency on missions which may put off some users.


Mod that can bring real added value to side missions:
[_Is only one example among many, but opens a door to real possible diversity_]


   [=Eden Objects=](https://steamcommunity.com/sharedfiles/filedetails/?id=686802825)


**1-** Journalist recovery mission on a helicopter crash

![Epave_KA60](https://user-images.githubusercontent.com/65271259/174707974-2b01e73e-e333-4ebd-90ae-b84580c6df8c.PNG)


**2-** Specific information recovery mission in an enemy dormitory

![Coffre_CSAT](https://user-images.githubusercontent.com/65271259/174708057-f31b52a2-d649-4148-af73-d44712b6c755.PNG)


**3-** Piloting a drone mission for the destruction of a specific objective

![Station_Drone](https://user-images.githubusercontent.com/65271259/174708224-9446a26c-591a-4472-a190-3f4533571da4.PNG)


**4-** Demining mission of a part of road

![Danger_Mine](https://user-images.githubusercontent.com/65271259/174708412-f4fcd321-e176-45e5-b48b-f5d1a7d78afa.PNG)

**Comment by arma-fury-road** [comment]
### Idea for adding option: linked to a mod from the Workshop


* [+] Brings a freshness and a renewal with the contribution of new elements in the options of the mission and sets up a collaboration with a creator of the Workshop.
* [-] Brings a dependency on missions which may put off some users.


Mod that can bring real added value to the mission options:
[_Is only one example among many, but opens a door to real possible diversity_]



   [=BTC_MoneyStacks=](https://steamcommunity.com/sharedfiles/filedetails/?id=1985865318)


**-** Option of being able to compensate civilians with money and recover credit from them as this can be done IRL
[_To be determined : how the Blufors recover this money. By setting in param.hpp or performing missions or both_]
![BTC_Money](https://user-images.githubusercontent.com/65271259/174710793-b57b8086-cc3c-438b-8ce4-f572ae22dd5e.PNG)

**Comment by PulsarNeutronStar** [comment]
Parameter: Distance SideMissions

Having Sidemissions partly on the other side of the Map can sometimes be hindering. It would be great if SideMissions could be, defined by a parameter, only spawn in acertain distance from the Main Base and established FOBs.

On top of that, a toggle to prioritise Towns that are not liberated/aligned would also be beneficial.

**Comment by SgtKrunch** [comment]
> Thanks to @mrNo0b , I come up with an idea.
> 
> Just use this inside the trigger:
> 
> ```
> [[btc_base]] call btc_fnc_city_cleanUp;
> if (btc_debug) then {
>     ["Base Cleanup Triggered."] remoteExec ["systemChat", 0];
> };
> ```
> 
> When trigger activates, the `btc_fnc_city_cleanUp` is called, and garbage within certain range of `btc_base` would be deleted, and this is fully compatible with the mission, without any need to think about item types and `btc_veh` or something.

@GoldJohnKing is this still working for you? I used to use it all the time. Came back to arma after a long time and it's no longer working anymore.

**Comment by GoldJohnKing** [comment]
> > Thanks to @mrNo0b , I come up with an idea.
> > Just use this inside the trigger:
> > ```
> > [[btc_base]] call btc_fnc_city_cleanUp;
> > if (btc_debug) then {
> >     ["Base Cleanup Triggered."] remoteExec ["systemChat", 0];
> > };
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > When trigger activates, the `btc_fnc_city_cleanUp` is called, and garbage within certain range of `btc_base` would be deleted, and this is fully compatible with the mission, without any need to think about item types and `btc_veh` or something.
> 
> @GoldJohnKing is this still working for you? I used to use it all the time. Came back to arma after a long time and it's no longer working anymore.

I am not sure, I have been away from Arma for a long time... I believe it works at that moment.

Before my AFK, I have moved to a mod called Advanced Garbage Collector to solve the garbage collection issue. It's much more convenient and much more effective. All you need to do is to put some of the units spawned by H&M into the mod's whitelist. I strongly suggest you give a try.

**Comment by Vdauphin** [comment]
Please focus on feature requests here
Do not hesitate to open discussion here: https://github.com/Vdauphin/HeartsAndMinds/discussions

**Comment by Kiskadee-dev** [comment]
Convoys could have a delay or depart at a specific time. Currently there's no time to plan anything especially if the path is short.

**Comment by Kiskadee-dev** [comment]
EMP devices could jam ACRE and TFAR radios

**Comment by KrisTheBest** [comment]
- Specific civilian roles (like a priest, a mayor, etc)
- Friendly militia (or police) in conquered cities
- Voice sounds during chatting
- Random sounds in cities 
- Random events in cities (like ceremonies in churches, markets, etc)

**Comment by STyx2909** [comment]
There is a need to list side missions on the presentation document.
So, not sure it is implemented or not:
- resupplying civilians (any supply, food, water, medical, tools...);
- repairing a civilian device;
- setting up a medical temporary tent to cure civilians;
- etc.

FYI a unit I was playing with was claiming defusing IEDs was not worth compared to missions and secondary objectives.
Dunno how it should be fixed without breaking current balance.

**Comment by STyx2909** [comment]
Not sure this MOD; "Guilt & Remembrance" is compatible. Could it be?
It is anyway a neat addition.

**Comment by STyx2909** [comment]
One request I would appreciate about initial load out of players.
Currently, there are 2 options:
- random loadout;
- no weapons.

A third choice: **Use defined loadouts in the mission** in the case you have edited them all.

**Comment by STyx2909** [comment]
What about using tickets to request AI support on players side?
We were 3 to play yesterday, and were obvioudly outnumbered.
I had to get in Zeus mode to spawn riflemen to give us a hand, twice.
I am not fond of this. Would have rather appreciated an in-game mechanic to perform the same action with a context.

**Comment by STyx2909** [comment]
I suggest to limit the number of side missions you can abort by setting a cost in ticket.
It is somewhat too easy to abort missions until you find one fitting your need.
Of course, we must ensure players team has requested assets to perform it (engineer, medic, etc.), in that case abortion is obvious.

**Comment by STyx2909** [comment]
Being the ones to mainly help population doesn't mean we are alone.
Possibility to add allied faction, in a limited and low number, to add animations and context would be great.

Thinking about, a single compound on the map, either (one at a time), infantry patrol, one vehicle patrol, road block, helicopter patrol (very rare).

Could be AAF on Altis and Malden or Gendarmerie.

**Comment by STyx2909** [comment]
Possibility to define high value objectives to conquer/occupy for Oplitas/players when setting up the game in Eden.
For example: solar plant on Altis.

**Comment by STyx2909** [comment]
> 

I know there is already something implemented but not for such type of objectives.

**Comment by STyx2909** [comment]
Would it be possible to "reward" in some way recovery of dead players bodies in the water?
Very awkward to perform and mostly impossible to achieve.
Thinking about something like any body put in a body bag below a certain depth can be counted as a valid ticket.

Also, in the same vein, placing dead civilians bodies in body bags, could very slightly improves relationship.

**Comment by STyx2909** [comment]
Would you mind turn "NATPO" and "Hoplitas" into variables one can change easily?
The idea behind it is for example to have "NATO" displayed as "UN" in texts and "Hoplitas" as "FIA".

**Comment by STyx2909** [comment]
More options to tune the Hoplitas:
- planes yes/no
- helicopters yes/no
Keeping it to the generic list of vehicles from Arma, knowing CARS does regroup a lot of different wheeled stuff.

**Comment by STyx2909** [comment]
Something that just happened to me.
I was stuck with the respawning CH-47I. It was partially damaged, but flyable.
I had to log off, to turn an Add On feature off to take off: APU-Start Up for the record.
When I logged in, the CH47 was respawned and in perfect condition, and I was also in base.
It would be better - IMHO - if the state and the position of the vehicle was saved (including content, there were 2 bodybags in this one) so it remains in place, without respawning. Idelly, player would respawn as well at the current location.

**Comment by STyx2909** [comment]
> 

Considering this feature, I have added such strongpoint to the map. Currently there are only 2 options: occupied or not occupied.
A  "Random" option might be interesting, so we don't know if they are occupied or not.

**Comment by STyx2909** [comment]
Not sure it is included, I have a hard time finding back links for a complete game documentation 🤪

But, shouldn't captured oplitas vehicles and claimed at the container, bring back some reputation?
Unarmed wheeled vehicle, 5 points
Armed wheeled vehicle, 10 points
Unarmed tracked vehicle 15 points (are there any???)
Armed tracked vehicle 20 points
Helicopter/plane 25 points

As a wreck, half the value. Rounded down.

**Comment by STyx2909** [comment]
Follow-up to the previous feature request about captured enemy vehicles, but this time if players are deciding to keep them instead of claiming reputation from them.
Considering they definitely lost vehicles and are unable to recover them from wrecks, they could use such vehicles, as in Liberation.
So far, my experience is pointing captured vehicles are despawned on restart/reloging.

**Comment by STyx2909** [comment]
Wrecks and bodies decay.

Bodies are removed after 5 in-game days
Wrecks are removed after 10 in-game days

**Comment by STyx2909** [comment]
Considering giving a look into this as a role-playing aspect of the game, possibility to display on a computer screen:
- strength of oplitas (weak, average, strong);
- intel level, based upon number of data have been gathered;
- population reputation;
- number of tickets (when shared)

Ideally looking to have tasks with map displayed on a large screen for briefing.

**Comment by STyx2909** [comment]
Splitting starting base and load-out configurations apart of the main mission.sqm so it is not overwritten when updating the version.
Certainly hard or impossible to do, but would be appreciated.
I can of course, copy and paste it into the new mission file, but it would avoid accidentally deleting it.

Good work on this new iteration. Can't way to test it.

**Comment by STyx2909** [comment]
Was about to add this to feature request for a while but never did:
An "easy" way to be able, by commenting code, to have a barebone version of Hearts and Minds mechanics. I mean here, the possibility to use specific towing, slinging, ViV, logistic MODs as far as it only concerns players side of course.

**Comment by PulsarNeutronStar** [comment]
Hey
would it be possible to add a "disable arsenal" option to btc_p_arsenal_Type?


My unit want to use a customized arsenal that we made and right now, we have to go into the core modules and disable it manually everytime we change maps.

**Comment by PulsarNeutronStar** [comment]
Parameter Option for Sidemissions: max distance from Base & Fobs

**Comment by STyx2909** [comment]
Could there be a way to have each version of Hearts and Minds to have their own parameters/database so we can run 2 versions in parallel on one server?
The best would be to save db into a specific directory so anyone willing to move on their campaign to a new verdion just have to move saved files into it.

**Comment by STyx2909** [comment]
I admit I am not a huge fan of unlimited resources. My bad.
So, anything that could increase gameplay in that field by using Arma 3 cost features would be much appreciated.

**Comment by PulsarNeutronStar** [comment]
Update Factions to include new Global Mobilisation Rebels.

**Comment by cmbtnt** [comment]
Would it be possible to improve the spawning for Hideouts? On Tanoa I've noticed that about half of the Hideouts spawn on the beach.

**Comment by Darkseth93** [comment]
Hey! I would like to suggest that side missions can be created for specific towns. eg. I want to do some things in Katkoula to bump my reputation.

**Comment by RoqueTHEGAMER** [comment]
Suggestion: Add ability for players to remove the rubbish themselves whiteout needing the vanilla vehicle, here is a code I am using on the OnPlayerRespawn.sqf so they can remove it after the IED is removed, the action also goes faster if it has a entrenching tool equipped, we should also get points with the civilians for that.

```sqf
[
	player,											// Object the action is attached to
	"Remove Rubbish",										// Title of the action
	"\a3\missions_f_oldman\data\img\holdactions\holdAction_box_ca.paa",	// Idle icon shown on screen
	"\a3\missions_f_oldman\data\img\holdactions\holdAction_box_ca.paa",	// Progress icon shown on screen
	"((getModelInfo cursorObject select 1) in btc_model_ieds) and {_this distance cursorObject < 3} and isSimpleObject cursorObject",						// Condition for the action to be shown
	"_caller distance cursorObject < 3",						// Condition for the action to progress
	{
		if ("ACE_EntrenchingTool" in (Items player)) then {_duration = 5};
	},													// Code executed when action starts
	{},													// Code executed on every progress tick
	{
	_minDistance = 3;
	_vehicle = player;

    private _ieds = allSimpleObjects [] - allSimpleObjects btc_type_blacklist;
    _ieds = (_ieds inAreaArray [getPosWorld _vehicle, 50, 50]) select {
        !("ace_drop" in ((getModelInfo _x) select 0))
    };
    if (_ieds isEqualTo []) exitWith {};

    _ieds = _ieds apply {[_x distance _vehicle, _x]};
    _ieds sort true;

    (_ieds select 0) params ["_distance", "_ied"];
    (0 boundingBoxReal _ied) params ["_p1", "_p2"];
    private _maxWidth = abs ((_p2 select 0) - (_p1 select 0));
    private _maxLength = abs ((_p2 select 1) - (_p1 select 1));
    if (
        _distance < (_minDistance + (_maxWidth max _maxLength) / 2) &&
        {allMines inAreaArray [getPosWorld _ied, 2.5, 2.5] isEqualTo []}
    ) then {
        private _pos = getPosATL _ied;
        _ied call CBA_fnc_deleteEntity;
        [btc_rep_bonus_IEDCleanUp, player] remoteExecCall ["btc_fnc_rep_change", 2];
        ["btc_ied_deleted", [_pos, player]] call CBA_fnc_serverEvent;
    } else {systemChat "Remove the IED first!"};
	},				// Code executed on completion
	{},													// Code executed on interrupted
	[],													// Arguments passed to the scripts as _this select 3
	12,													// Action duration [s]
	10,													// Priority
	false,												// Remove on completion
	false												// Show in unconscious state
] call BIS_fnc_holdActionAdd;
```

**Comment by RoqueTHEGAMER** [comment]
In line 328 of core\def\mission.sqf at btc_type_tags_sentences we could add our own sentences but it will return a empty string because is not localized, that can be easily fixed by adding a check to see if is a empty string from the localization, if it is then just use the full string.
here is a exemple for initArea.sqf in line 67
```sqf
private _SelectedSetences = (selectRandom btc_type_tags_sentences);
private _sentences = ((localize _SelectedSetences ) splitString " ");
If (count _sentences == 0) then {_sentences=_SelectedSetences splitString " "};
```

**Comment by TheLocalPub** [comment]
> Suggestion: Add ability for players to remove the rubbish themselves whiteout needing the vanilla vehicle, here is a code I am using on the OnPlayerRespawn.sqf so they can remove it after the IED is removed, the action also goes faster if it has a entrenching tool equipped, we should also get points with the civilians for that, its also a problem when there is rubbish in the middle of the road complicating logistics and causing accidents with civilian cars hitting those things.
> 
> ```sqf
> [
> 	player,											// Object the action is attached to
> 	"Remove Rubbish",										// Title of the action
> 	"\a3\missions_f_oldman\data\img\holdactions\holdAction_box_ca.paa",	// Idle icon shown on screen
> 	"\a3\missions_f_oldman\data\img\holdactions\holdAction_box_ca.paa",	// Progress icon shown on screen
> 	"((getModelInfo cursorObject select 1) in btc_model_ieds) and {_this distance cursorObject < 3} and isSimpleObject cursorObject",						// Condition for the action to be shown
> 	"_caller distance cursorObject < 3",						// Condition for the action to progress
> 	{
> 		if ("ACE_EntrenchingTool" in (Items player)) then {_duration = 5};
> 	},													// Code executed when action starts
> 	{},													// Code executed on every progress tick
> 	{
> 	_minDistance = 3;
> 	_vehicle = player;
> 
>     private _ieds = allSimpleObjects [] - allSimpleObjects btc_type_blacklist;
>     _ieds = (_ieds inAreaArray [getPosWorld _vehicle, 50, 50]) select {
>         !("ace_drop" in ((getModelInfo _x) select 0))
>     };
>     if (_ieds isEqualTo []) exitWith {};
> 
>     _ieds = _ieds apply {[_x distance _vehicle, _x]};
>     _ieds sort true;
> 
>     (_ieds select 0) params ["_distance", "_ied"];
>     (0 boundingBoxReal _ied) params ["_p1", "_p2"];
>     private _maxWidth = abs ((_p2 select 0) - (_p1 select 0));
>     private _maxLength = abs ((_p2 select 1) - (_p1 select 1));
>     if (
>         _distance < (_minDistance + (_maxWidth max _maxLength) / 2) &&
>         {allMines inAreaArray [getPosWorld _ied, 2.5, 2.5] isEqualTo []}
>     ) then {
>         private _pos = getPosATL _ied;
>         _ied call CBA_fnc_deleteEntity;
>         [btc_rep_bonus_IEDCleanUp, player] remoteExecCall ["btc_fnc_rep_change", 2];
>         ["btc_ied_deleted", [_pos, player]] call CBA_fnc_serverEvent;
>     } else {systemChat "Remove the IED first!"};
> 	},				// Code executed on completion
> 	{},													// Code executed on interrupted
> 	[],													// Arguments passed to the scripts as _this select 3
> 	12,													// Action duration [s]
> 	10,													// Priority
> 	false,												// Remove on completion
> 	false												// Show in unconscious state
> ] call BIS_fnc_holdActionAdd;
> ```

How'd this go?
Does it work fine without issues, multiplayer compatible with JIPs also?

**Comment by TheLocalPub** [comment]
A few suggestions:
- An option in the parameters for unlimited respawn.
- An option in the parameters to disable ticket deduction upon vehicle respawning (Potentially expand this to disable, decrease, or increase ticket lose from a vehicle based on it's type, i,e helicopters or tanks would cost more tickets.)
- Place a trigger or marker at spawn in the example mission that when players enter the area or are within X radius of the an object, they get full ACE healed using `ace_medical_fnc_treatmentAdvanced_fullHealLocal;`

**Comment by STyx2909** [comment]
> A few suggestions:
> 
>     * Place a trigger or marker at spawn in the example mission that when players enter the area or are within X radius of the an object, they get full ACE healed using `ace_medical_fnc_treatmentAdvanced_fullHealLocal;`

You can create the trigger yourself and use following settings:
- Activation:
Any player or BLUFORCE or whatever suits you
- Activation type:
present
- repeatable:
on
- Trigger expression: 
call{this}
- and into the On activation field:
call{{[_x,_x] call ace_medical_treatment_fnc_fullHealLocal;} forEach thisList;}

**Comment by UnRealxInferno** [comment]
Can a zone blacklist be added within 1km of the HQ? I want to use an airfield as a base but it spawns opfor on the airfield and there's seemingly no way to stop them?

**Comment by TheLocalPub** [comment]
> Can a zone blacklist be added within 1km of the HQ? I want to use an airfield as a base but it spawns opfor on the airfield and there's seemingly no way to stop them?

Yes.
See the link. 
https://vdauphin.github.io/HeartsAndMinds/Prevent-AI-from-spawning-near-the-base

**Comment by STyx2909** [comment]
Reputation gain when IED is destroyed with Demolition blocks, without being defused.
Could not find that ACE interaction location, so I blew up the IED with some C4.

**Comment by deBearN** [comment]
> Reputation gain when IED is destroyed with Demolition blocks, without being defused. Could not find that ACE interaction location, so I blew up the IED with some C4.

That or either a controlled explosion without causing civ_hd/killed, building damaged/destroyed, player killed, my unit almost never defuses IED's apart from the side quest ones, we just prepare a controlled explosion by shooting at it and making sure there are no civs nearby

**Comment by STyx2909** [comment]
> > Reputation gain when IED is destroyed with Demolition blocks, without being defused. Could not find that ACE interaction location, so I blew up the IED with some C4.
> 
> That or either a controlled explosion without causing civ_hd/killed, building damaged/destroyed, player killed, my unit almost never defuses IED's apart from the side quest ones, we just prepare a controlled explosion by shooting at it and making sure there are no civs nearby

I agree, though defusing IEDs should be more rewarding than destroying them at first. Better reputation gain for defusing. A small gain for destroying them.

**Comment by Daek-wml** [comment]
> Suggestion: Add ability for players to remove the rubbish themselves whiteout needing the vanilla vehicle, here is a code I am using on the OnPlayerRespawn.sqf so they can remove it after the IED is removed, the action also goes faster if it has a entrenching tool equipped, we should also get points with the civilians for that, its also a problem when there is rubbish in the middle of the road complicating logistics and causing accidents with civilian cars hitting those things.
> 
> ```sqf
> [
> 	player,											// Object the action is attached to
> 	"Remove Rubbish",										// Title of the action
> 	"\a3\missions_f_oldman\data\img\holdactions\holdAction_box_ca.paa",	// Idle icon shown on screen
> 	"\a3\missions_f_oldman\data\img\holdactions\holdAction_box_ca.paa",	// Progress icon shown on screen
> 	"((getModelInfo cursorObject select 1) in btc_model_ieds) and {_this distance cursorObject < 3} and isSimpleObject cursorObject",						// Condition for the action to be shown
> 	"_caller distance cursorObject < 3",						// Condition for the action to progress
> 	{
> 		if ("ACE_EntrenchingTool" in (Items player)) then {_duration = 5};
> 	},													// Code executed when action starts
> 	{},													// Code executed on every progress tick
> 	{
> 	_minDistance = 3;
> 	_vehicle = player;
> 
>     private _ieds = allSimpleObjects [] - allSimpleObjects btc_type_blacklist;
>     _ieds = (_ieds inAreaArray [getPosWorld _vehicle, 50, 50]) select {
>         !("ace_drop" in ((getModelInfo _x) select 0))
>     };
>     if (_ieds isEqualTo []) exitWith {};
> 
>     _ieds = _ieds apply {[_x distance _vehicle, _x]};
>     _ieds sort true;
> 
>     (_ieds select 0) params ["_distance", "_ied"];
>     (0 boundingBoxReal _ied) params ["_p1", "_p2"];
>     private _maxWidth = abs ((_p2 select 0) - (_p1 select 0));
>     private _maxLength = abs ((_p2 select 1) - (_p1 select 1));
>     if (
>         _distance < (_minDistance + (_maxWidth max _maxLength) / 2) &&
>         {allMines inAreaArray [getPosWorld _ied, 2.5, 2.5] isEqualTo []}
>     ) then {
>         private _pos = getPosATL _ied;
>         _ied call CBA_fnc_deleteEntity;
>         [btc_rep_bonus_IEDCleanUp, player] remoteExecCall ["btc_fnc_rep_change", 2];
>         ["btc_ied_deleted", [_pos, player]] call CBA_fnc_serverEvent;
>     } else {systemChat "Remove the IED first!"};
> 	},				// Code executed on completion
> 	{},													// Code executed on interrupted
> 	[],													// Arguments passed to the scripts as _this select 3
> 	12,													// Action duration [s]
> 	10,													// Priority
> 	false,												// Remove on completion
> 	false												// Show in unconscious state
> ] call BIS_fnc_holdActionAdd;
> ```

How are you applying this? I tried it in the editor as a multiplayer host, and it works. However, when hosted on a dedicated server via FASTER, the addAction doesn’t work.

**Comment by deBearN** [comment]
Option for a third party, temporarily friendly or otherwise, dealing with insurgencies and/or armed military will most of the times result in 3 parties involved in the conflict

YourFaction, Insurgency1, Insurgency2, they are not necessarily friendly to each other, but could be
YourFaction, LocalGov, Insurgency, like what happens at the lebanon
YourFaction, LocalGov1, LocalGov2, stuff like what happened at sudan where after a coup 2 parts of the military fight each other


this could lead to more interesting scenarios, like either make the LocalGov enemy by default or tie them to the civillian reputation so if we anger civs it will also anger the Occupying Governamental faction and make them hostile, and as a peace making mission you are assisting whatever faction the UN currently supports for example, like the 2 factions Sahrani lore has

Having an param to add a 3rd party
by enabling it enables options to 
- Choose if they're partially-friendly or not
- They are allied to the Oplitas or not
- If friendly, they can be tied to the reputation system or not

great options for peace keeping scenarios

**Comment by JKaiosilva** [comment]
During the short time I spent in the Brazilian Navy, I noticed that certain roles are performed only by specific military ranks.
My suggestion would be to create a ranking option in the game, where players could gain experience and good reputation to achieve a higher rank.

Example:
1 - Recruit - Everyone starts with this rank. After picking up a loadout or completing a simple verification task on the server, they would be promoted to Soldier.
2 - Soldier - Able to use all basic equipment and simple vehicles for up to 2 players.
3 - Corporal - A more experienced soldier capable of using more specific equipment. (As in the military, where Corporals specialize in a specific role in warfare.) They can repair vehicles, use heavier weaponry, and geolocation equipment.
4 - Sergeant - An experienced commander, acting as the right-hand of the game's high command, guiding troops, and using all terrestrial vehicles, weaponry, and advanced medical equipment. (Within the Sergeant rank, there could be three subdivisions to increase the sense of progression.)
5 - Lieutenant - An official of the server, responsible for a specific area of operation in the game, such as terrestrial vehicles, aircraft, or training new players. (This rank would not require specific configuration in the game, as it would be a recommendation for each server to use as they wish.) The Lieutenant rank would be granted by the server owner, as it is a position of trust, where the player represents the high command within the game.
6 - General - Granted to the server owner, who alone can configure the server.

If you would like assistance in further developing these rank divisions, I am available to help. Thank you in advance. Best regards!

**Comment by colt55** [comment]
Feature Request – Score-Based Arsenal Progression + Player Ranking System

Request Summary:

We’d like to propose a modular system that ties a player’s score (tracked via profileNamespace) to both Arsenal access and a personal rank system. This enhances realism and long-term engagement without needing persistent reputation or admin oversight.

Feature Modules:

 Score-Based Arsenal Unlocks
Restrict ACE Arsenal gear by score tiers. Gear classes unlock as the player’s personal score increases:

Tier 0 (0–49): Light rifles (e.g., AK-74), basic gear, bandages

Tier 1 (50–99): Better rifles (AK-103), pistol, IV kits

Tier 2 (100–199): DMRs, optics, radios

Tier 3 (200+): Launchers, advanced equipment, range gear

Death Penalty
Player death subtracts a configurable score value (e.g., -10). Optional cap so score doesn’t drop below 0. Optional: Friendly fire penalty.

 Anti-Exploit Gear Filter
Prevent players from looting or equipping gear above their current score tier.
Possible methods:

Arsenal filtering

Loadout validation on respawn

Hook into Take or InventoryClosed events

Score-Based Player Rank System
Assign a visible rank name based on player score.
Example:

sqf
Copy
Edit
if (_score < 50) then {_rank = "Recruit"};
else if (_score < 100) then {_rank = "Private"};
else if (_score < 200) then {_rank = "Corporal"};
else if (_score < 300) then {_rank = "Sergeant"};
else {_rank = "Veteran"};
Rank could be shown in side chat, ACE name tag, or via a hint on login

Optional setting to display rank on scroll action ("Check Status")

 Mission Parameter Controls (Optional)
Toggle components:

Enable/disable Arsenal filtering

Enable/disable death penalty

Enable/disable rank display

Why This Helps:

Hearts and Minds is already immersive, but this adds personal progression and role value to gameplay. It’s especially good for public servers or solo ops where player reputation and admin promotion systems don’t exist.

We’ve prototyped parts of this system in our version and would love to see an official, extensible integration.

S!

**Comment by colt55** [comment]
Feature Request: Improvements to Ticket System (Anti-Cheat & Save Issues)

Hey guys,

First, thanks for keeping Hearts and Minds alive all these years. Solid work, even if you occasionally break more stuff than you fix—classic Arma modding, right?
I’d like to request a feature (or at least some fixes) for the respawn ticket system. We’re running a public server and having a hell of a time keeping cheaters and exploiters from abusing the ticket claim system. People are spamming tickets, sometimes grabbing more than they should, and occasionally using workarounds to reset or dupe them.
On top of that, the current system sometimes fails to properly save ticket data after a server restart or mission reload, which makes running long-term campaigns an exercise in futility.
- Here’s what we’re running into:
- Players can spam or exploit the ticket claim action, getting more lives than they should.
- Tickets sometimes don’t properly save after a restart, even when most other mission progress is kept.
- Would be great to have an anti-spam/delay feature, and/or some kind of server-side save/restore for tickets tied to player UID.
- Bonus points if there’s a clean way to log or catch people blatantly abusing the system.
If you’ve got a fix or any planned changes, let us know. If you need a tester who’s been around since OFP, I’m happy to help break things for you.
And if you’re going to tell me “PR for review” or “patch welcome,” at least make the ticket system a little less 2004 first.

**Comment by UnRealxInferno** [comment]
Please add support for ATLAS/AEGIS/REVOLUCION
https://steamcommunity.com/sharedfiles/filedetails/?id=2684335735

**Comment by CLAFMB** [comment]
Hi, would it be possible to add this mod so that it can be used in Hearts & Minds?

https://steamcommunity.com/sharedfiles/filedetails/?id=735566597

**Comment by deBearN** [comment]
> Hi, would it be possible to add this mod so that it can be used in Hearts & Minds?
> 
> https://steamcommunity.com/sharedfiles/filedetails/?id=735566597

in the [Hearts and minds documentation](https://vdauphin.github.io/HeartsAndMinds/), there's a place for you to see how to add new factions

**Comment by UnRealxInferno** [comment]
Requesting support for https://steamcommunity.com/sharedfiles/filedetails/?id=2412164804&searchtext=r3f for a better, more persistent building/logistics system.

**Comment by axn14** [comment]
Can we add IEDD notebooks to the IED framework in game? makes defusal more interesting since it includes proximity detonation, countdown based, VBIEDs that are speed based and then suicide bombers as well:
https://steamcommunity.com/workshop/filedetails/?id=3048818056

If not, please advice how i can help to add this into the game mode to not just have the ace style defusal?
